onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -2527,6 +2527,20 @@ static constexpr OrtApi ort_api_1_to_12 = {
     &OrtApis::CreateOp,
     &OrtApis::InvokeOp,
     &OrtApis::ReleaseOp,
+#ifdef ENABLE_TRAINING_ON_DEVICE
+    // Experimental for on-device training. Always keep at the bottom.
```

Comment:
Since the training APIs are experimental can they live in a separate header? They don't need to be subjected to the same versioning constraints as the inferencing APIs; you can even skip versioning altogether if things are experimental.